pkgs/development/libraries/qt-6/functions/broken-mkspecs-files-disable.sh

```diff
@@ -0,0 +1,53 @@
+# https://github.com/NixOS/nixpkgs/issues/214765
+
+# the functions qtPrepareTool and qtPrepareLibExecTool are defined in
+# qtbase-dev/mkspecs/features/qt_functions.prf
+
+brokenMkspecsFilesDisable() {
+  # check if a tool is installed in the expected location.
+  # if the tool is not installed, disable the mkspecs file
+  # by moving it to *.bak.
+  # the disabled mkspecs files will be restored by
+  # restoreMkspecsFiles in downstream derivations, for example qttools.
+  local source_prefix=$1 # $dev
+  local target_prefix=$2 # $dev
+  local instloc_expected=$3 # $dev/bin or $dev/libexec
+  local funcname_expected=$4 # qtPrepareTool or qtPrepareLibExecTool
+  local mkspecs_path=mkspecs/features
+  local mkspecs_file_pattern='*.prf'
+  local debug=${DEBUG:-false}
+  local dryrun=${DRYRUN:-false}
+  local dryrun_cmd=
+  if $dryrun; then dryrun_cmd=echo; fi
```

Comment:
Maybe `NIX_DRYRUN`? Do we really need the dryrun functionality, it appears to be a duplicate of DEBUG.

Reply:
dryrun is useful for debugging and unit-testing